src/Seq.Input.HealthCheck/HttpHealthCheck.cs

```diff
@@ -124,7 +125,74 @@ public async Task<HealthCheckResult> CheckNow(CancellationToken cancel)
                 initialContent,
                 exception,
                 data,
-                _targetUrl == probedUrl ? null : probedUrl);
+                _targetUrl == probedUrl ? null : probedUrl,
+                redirectCount: _redirectedUrls.Count,
+                finalUrl: finalUrl);
+        }
+
+        private void AddHeadersToRequest(HttpRequestMessage request, string probeId)
+        {
+            request.Headers.Add(CorrelationHeaderId, probeId);
+            foreach (var (name, value) in _headers)
+            {
+                // This will throw if a header is duplicated (better for the user to detect this configuration problem).
+                request.Headers.Add(name, value);
+            }
+
+            if (_bypassHttpCaching)
+                request.Headers.CacheControl = new CacheControlHeaderValue {NoStore = true};
+        }
+
+        private async Task<HttpResponseMessage> MakeAndFollowRequest(CancellationToken cancel, string requestUri, string correlationId,
+            Stack<Redirect> previousRedirects)
+        {
+            var request = new HttpRequestMessage(HttpMethod.Get, requestUri);
+            // include initial headers on every redirect
+            // todo: Should any response headers included in future requests?
+            AddHeadersToRequest(request, correlationId);
```

Comment:
Probably not response headers, but there will be some cases in which people will want cookies from the response to be forwarded with subsequent requests. May be scope creep, perhaps this is one for a later/future feature?